tests/subtractions/test_db.py

```diff
@@ -1,61 +1,83 @@
-import pytest
-
 import virtool.subtractions.db
 from virtool.data.transforms import apply_transforms
+from virtool.fake.next import DataFaker
 from virtool.subtractions.db import (
     AttachSubtractionsTransform,
     unlink_default_subtractions,
 )
+from virtool.uploads.models import UploadType
+
 
+async def test_attach_subtractions(fake: DataFaker, mongo, snapshot):
+    user = await fake.users.create()
+    upload = await fake.uploads.create(
+        user=user, upload_type=UploadType.subtraction, name="foobar.fq.gz"
+    )
 
-@pytest.mark.parametrize(
-    "documents",
-    [
-        {"id": "sub_1", "subtractions": ["foo", "bar"]},
+    subtraction_ids = [
+        subtraction.id
+        for subtraction in [
+            await fake.subtractions.create(user=user, upload=upload) for _ in range(2)
+        ]
+    ]
+
+    documents = [
+        {"id": "sub_1", "subtractions": subtraction_ids},
         [
-            {"id": "sub_1", "subtractions": ["foo", "bar"]},
-            {"id": "sub_2", "subtractions": ["foo"]},
+            {"id": "sub_1", "subtractions": subtraction_ids},
+            {"id": "sub_2", "subtractions": [subtraction_ids[0]]},
             {"id": "sub_3", "subtractions": []},
         ],
-    ],
-)
-async def test_attach_subtractions(documents, mongo, snapshot):
-    await mongo.subtraction.insert_many(
-        [{"_id": "foo", "name": "Foo"}, {"_id": "bar", "name": "Bar"}], session=None
-    )
+    ]
+
+    for document in documents:
+        result = await apply_transforms(document, [AttachSubtractionsTransform(mongo)])
 
-    result = await apply_transforms(documents, [AttachSubtractionsTransform(mongo)])
+        assert result == snapshot
```

Comment:
The way this was before was faster.

Reply:
Not sure how to pass the created subtraction Id to the parameterized test. I guess I should just write a separate function to pass to the `pytest.mark.parametrize` so the tests can be parallelized?

Reply:
Do you want me to break this up into multiple tests? There is no straight forward way to get code execution within the `pytest.mark.parametrize` that works nicely with the new fake API.

Reply:
This test isn't parametrized is it?

I am just saying the last few lines can look like this instead:
```py
assert await apply_transforms(document, [AttachSubtractionsTransform(mongo)]) == snapshot
```

The `apply_transforms` function can take either single document or list of documents as its first argument.